src/it/java/ar/com/carloscurotto/storm/complex/topology/LocalStormTopologyIT.java

```diff
@@ -0,0 +1,106 @@
+package ar.com.carloscurotto.storm.complex.topology;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.activemq.broker.BrokerService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import ar.com.carloscurotto.storm.complex.model.Result;
+import ar.com.carloscurotto.storm.complex.model.Update;
+import ar.com.carloscurotto.storm.complex.model.UpdateRow;
+import ar.com.carloscurotto.storm.complex.service.UpdateService;
+import backtype.storm.Config;
+import backtype.storm.ILocalCluster;
+import backtype.storm.Testing;
+import backtype.storm.generated.AlreadyAliveException;
+import backtype.storm.generated.InvalidTopologyException;
+import backtype.storm.testing.MkClusterParam;
+import backtype.storm.testing.TestJob;
+import backtype.storm.utils.Utils;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({ "/app-config.xml" })
+public class LocalStormTopologyIT {
+
+    private static final String BROKER_URL = "tcp://localhost:61616";
+
+    private BrokerService brokerService;
+
+    @Autowired
+    private UpdateService updateService;
+
+    @Autowired
+    private UpdateTopologyConfiguration updateTopologyConfiguration;
+
+    @Before
+    public void before() throws Exception {
+        brokerService = new BrokerService();
+        brokerService.setPersistent(false);
+        brokerService.setUseJmx(false);
+        brokerService.addConnector(BROKER_URL);
+        brokerService.start();
+    }
+
+    /**
+     * Encontre que Nathan hace referencia a testear la topology basandose en los siguientes ejemplos:
+     * https://github.com/xumingming/storm-lib/blob/master/src/jvm/storm/TestingApiDemo.java
+     *
+     */
+    @Test
+    public void test() {
+        MkClusterParam mkClusterParam = new MkClusterParam();
+        mkClusterParam.setSupervisors(2);
+        mkClusterParam.setPortsPerSupervisor(5);
+        final Config daemonConf = new Config();
+
+        Testing.withLocalCluster(mkClusterParam, new TestJob() {
+            @Override
+            public void run(final ILocalCluster cluster) throws AlreadyAliveException, InvalidTopologyException {
+                cluster.submitTopology("complex-updates", daemonConf, updateTopologyConfiguration.getStormTopology());
+                Update firstUpdate = createUpdateFor("id-1", "SEMS", "row-1");
+                //El sleep se debe a que hay un race condition entre el submit del mensaje y la topology que no termina de levantar
```

Comment:
La otra que se me ocurre es ver is hay un submit topology sincronico con un timeout pero no se si te lo da storm.